app/services/segment_delete_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class SegmentDeleteService
+  SEGMENT_URI = URI('https://gdpr.segment.com/graphql')
+
+  class SegmentDeleteServiceError < StandardError; end
+  class MissingAuthenticationData < SegmentDeleteServiceError; end
+  class SegmentAuthenticationError < SegmentDeleteServiceError; end
+  class SegmentUserDeletionError < SegmentDeleteServiceError; end
+
+
+  def initialize(user_id)
+    @user_id = user_id
+    @errors = ActiveModel::Errors.new(self)
+  end
+
+  attr_reader :user_id, :errors
+
+  def delete_user!
+    return unless Features::SegmentDeletionConfig.enabled?
+    request_body = {"query": "mutation { createWorkspaceRegulation(workspaceSlug: \"3scale\" type: SUPPRESS_AND_DELETE userId: \"#{user_id}\" ) { id } }"}.to_json
+    response = perform_request(request_body: request_body, authorization: token)
+    raise SegmentUserDeletionError, response.body unless response.code == '200'
```

Comment:
`response.code.to_s` to make it bulletproof.

Reply:
`response.code` already returns a String, but it won't harm.

Reply:
I don't think `response.body` should be the error message. It could be anything coming from an external service. Maybe better adding as metadata.

Reply:
From Gui when I asked him about this:
> oh! the message can be whatever you want
> but the response body should be in a parameter